lib/straight/order.rb

```diff
@@ -128,16 +128,17 @@ def get_transaction_status(reload: false, transactions: nil)
             [transaction(reload: reload)]
           end.compact
 
-        amount_paid = transactions.map { |t| t.amount }.reduce(:+) || 0
+        uniq_transactions = transactions.group_by(&:tid).each_value.map(&:first)
```

Comment:
Yeah, but explicit version gives more control over which transactions are picked.
Currenly, it's a `&:first`, but who knows, maybe some additional checks will be required.